app/ui/src/main/java/com/fsck/k9/ui/K9Drawer.java

```diff
@@ -64,24 +82,68 @@ public K9Drawer(MessageList parent, Bundle savedInstanceState) {
                 .withOnDrawerItemClickListener(createItemClickListener())
                 .withOnDrawerListener(parent.createOnDrawerListener())
                 .withSavedInstance(savedInstanceState)
+                .withAccountHeader(buildAccountHeader())
                 .build();
 
-        addHeaderItems();
         addFooterItems();
 
         initializeFolderIcons();
     }
 
-    private void addHeaderItems() {
+    private AccountHeader buildAccountHeader() {
+        AccountHeaderBuilder headerBuilder = new AccountHeaderBuilder()
+                .withActivity(parent)
+                //.withTextColorRes(R.color.primary_text_secondary_when_activated_material)
+                .withTranslucentStatusBar(false);
+
         if (!K9.isHideSpecialAccounts()) {
-            drawer.addItems(new PrimaryDrawerItem()
-                            .withName(R.string.integrated_inbox_title)
-                            .withIcon(getResId(R.attr.iconUnifiedInbox))
-                            .withIdentifier(DRAWER_ID_UNIFIED_INBOX),
-                    new DividerDrawerItem());
+            headerBuilder.addProfiles(new ProfileDrawerItem()
+                    .withNameShown(true)
+                    .withName(R.string.integrated_inbox_title)
+                    .withEmail(parent.getString(R.string.integrated_inbox_detail))
+                    .withIcon(new IconicsDrawable(parent, FontAwesome.Icon.faw_users)
+                            .colorRes(R.color.material_drawer_background).backgroundColor(Color.GRAY)
+                            .sizeDp(56).paddingDp(8))
+                    .withSetSelected(unifiedInboxSelected)
+                    .withIdentifier(DRAWER_ID_UNIFIED_INBOX)
+            );
+        }
 
-            headerItemCount += 2;
+        List <Account> accounts = preferences.getAccounts();
+        for (int i = preferences.getAccounts().size() - 1; i >= 0; i--) {
```

Comment:
Why are you adding accounts in reverse order?

Reply:
leftover from drawer item implementation...